#Base de datos de Empresas de Norte de Santander, Colombia
##Datos extraidos de la pagina de datos abiertos de colombia https://www.datos.gov.co/
Esta Base de Datos contiene la información de los empresarios registrado en la Cámara de Comercio de Cúcuta, la cual es captura al momento de la inscripción o renovación del registro mercantil por parte de los empresarios.(https://www.datos.gov.co/Comercio-Industria-y-Turismo/BASE-DE-DATOS-EMPRESAS-DE-NORTE-DE-SANTANDER/w98a-ssvz/about_data)

In [ ]:
import pandas as pd
url='Tasas_de_inter_s_activas_por_tipo_de_cr_dito_20240115.csv'
df=pd.read_csv(url)

In [ ]:
df.head()

Tipo_Entidad          Nombre_Tipo_Entidad  Codigo_Entidad   Nombre_Entidad  \
0             1  BC-ESTABLECIMIENTO BANCARIO               1  Banco de Bogotá   
1             1  BC-ESTABLECIMIENTO BANCARIO              52    Bancamía S.A.   
2             1  BC-ESTABLECIMIENTO BANCARIO              43       Banagrario   
3             1  BC-ESTABLECIMIENTO BANCARIO               7      Bancolombia   
4             1  BC-ESTABLECIMIENTO BANCARIO               1  Banco de Bogotá   

  Fecha_Corte Tipo_de_persona       Sexo Tamaño_de_empresa  \
0  20/10/2023         Natural  Masculino         No aplica   
1  20/10/2023         Natural  Masculino      Microempresa   
2  10/11/2023         Natural  Masculino         No aplica   
3  01/12/2023        Jurídica  No aplica   Mediana empresa   
4  24/11/2023         Natural   Femenino         No aplica   

       Tipo_de_crédito                                   Tipo_de_garantía  \
0              Consumo                                       Sin garantia   
1   Crédito productivo  Garantía  fondo nacional de garantías (FNG) o ...   
2              Consumo                                       Sin garantia   
3  Comercial ordinario                        Garantia idónea o no idónea   
4              Consumo                                       Sin garantia   

   ... margen_adicional Montos_desembolsados  \
0  ...              0.0           2197830.00   
1  ...              0.0           2800000.00   
2  ...              0.0           7969200.00   
3  ...              0.0           3538866.93   
4  ...              0.0            585000.00   

   Numero_de_creditos_desembolsados         Grupo_Etnico  \
0                                11  Sin información (1)   
1                                 1  Sin información (1)   
2                                 4  Sin información (1)   
3                                 5            No aplica   
4                                 1  Sin información (1)   

   Antiguedad_de_la_empresa  Tipo_de_Tasa  \
0              No aplica(1)            FS   
1            más de 10 años            FS   
2              No aplica(1)            FS   
3                0 a 5 años            FS   
4              No aplica(1)            FS   

                    Rango_monto_desembolsado                Clase_deudor  \
0                              Hasta 1 SMLMV        Deudor de la entidad   
1  Mayor a 2 SMLMV y menor o igual a 3 SMLMV        Deudor de la entidad   
2    Mayor a 1 SMLMV menor o igual a 3 SMLMV        Deudor de la entidad   
3    Mayor a 1 SMLMV menor o igual a 3 SMLMV        Deudor de la entidad   
4                              Hasta 1 SMLMV  Deudor nuevo en la entidad   

  Codigo_CIIU Codigo_Municipio  
0        10.0           5172.0  
1       143.0          47268.0  
2       150.0          41676.0  
3      8544.0          11001.0  
4      4711.0          70823.0  

[5 rows x 23 columns]

In [ ]:
df.columns

Index(['Tipo_Entidad', 'Nombre_Tipo_Entidad', 'Codigo_Entidad',
       'Nombre_Entidad', 'Fecha_Corte', 'Tipo_de_persona', 'Sexo',
       'Tamaño_de_empresa', 'Tipo_de_crédito', 'Tipo_de_garantía',
       'Producto de crédito', 'Plazo de crédito',
       'Tasa_efectiva_promedio_ponderada', 'margen_adicional',
       'Montos_desembolsados', 'Numero_de_creditos_desembolsados',
       'Grupo_Etnico', 'Antiguedad_de_la_empresa', 'Tipo_de_Tasa',
       'Rango_monto_desembolsado', 'Clase_deudor', 'Codigo_CIIU',
       'Codigo_Municipio'],
      dtype='object')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8443774 entries, 0 to 8443773
Data columns (total 23 columns):
 #   Column                            Dtype  
---  ------                            -----  
 0   Tipo_Entidad                      int64  
 1   Nombre_Tipo_Entidad               object 
 2   Codigo_Entidad                    int64  
 3   Nombre_Entidad                    object 
 4   Fecha_Corte                       object 
 5   Tipo_de_persona                   object 
 6   Sexo                              object 
 7   Tamaño_de_empresa                 object 
 8   Tipo_de_crédito                   object 
 9   Tipo_de_garantía                  object 
 10  Producto de crédito               object 
 11  Plazo de crédito                  object 
 12  Tasa_efectiva_promedio_ponderada  float64
 13  margen_adicional                  float64
 14  Montos_desembolsados              float64
 15  Numero_de_creditos_desembolsados  int64  
 16  Grupo_Etnico                      ob

In [ ]:
df.isnull().sum()

Tipo_Entidad                             0
Nombre_Tipo_Entidad                      0
Codigo_Entidad                           0
Nombre_Entidad                           0
Fecha_Corte                              0
Tipo_de_persona                          2
Sexo                                     2
Tamaño_de_empresa                        2
Tipo_de_crédito                          2
Tipo_de_garantía                         2
Producto de crédito                      4
Plazo de crédito                         2
Tasa_efectiva_promedio_ponderada         0
margen_adicional                         0
Montos_desembolsados                     0
Numero_de_creditos_desembolsados         0
Grupo_Etnico                        251462
Antiguedad_de_la_empresa            251462
Tipo_de_Tasa                        251462
Rango_monto_desembolsado            251462
Clase_deudor                        251462
Codigo_CIIU                         251462
Codigo_Municipio                    251462
dtype: int6

In [ ]:
df.shape

(8192306, 23)

Procedemos a eliminar las filas con valores nulos, aunque con bastantes comparado con la cantidad total, son suficientes para el analisis

In [ ]:
df = df.dropna()

comprobamos que se eliminaron los valores nulos

In [ ]:
df.isnull().sum()

Tipo_Entidad                        0
Nombre_Tipo_Entidad                 0
Codigo_Entidad                      0
Nombre_Entidad                      0
Fecha_Corte                         0
Tipo_de_persona                     0
Sexo                                0
Tamaño_de_empresa                   0
Tipo_de_crédito                     0
Tipo_de_garantía                    0
Producto de crédito                 0
Plazo de crédito                    0
Tasa_efectiva_promedio_ponderada    0
margen_adicional                    0
Montos_desembolsados                0
Numero_de_creditos_desembolsados    0
Grupo_Etnico                        0
Antiguedad_de_la_empresa            0
Tipo_de_Tasa                        0
Rango_monto_desembolsado            0
Clase_deudor                        0
Codigo_CIIU                         0
Codigo_Municipio                    0
dtype: int64

ahora verificaremos si existen valores duplicados

In [ ]:
df.duplicated().sum()

4

procedemos a borrar los duplicados

In [ ]:
df=df.drop_duplicates()

In [ ]:
df.shape

(8192306, 23)

antes de hacer la limpieza (8443774, 23), despues de borrar los valores nulos y duplicados (8192306, 23) borrando 251468 datos

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8192306 entries, 0 to 8443773
Data columns (total 23 columns):
 #   Column                            Dtype  
---  ------                            -----  
 0   Tipo_Entidad                      int64  
 1   Nombre_Tipo_Entidad               object 
 2   Codigo_Entidad                    int64  
 3   Nombre_Entidad                    object 
 4   Fecha_Corte                       object 
 5   Tipo_de_persona                   object 
 6   Sexo                              object 
 7   Tamaño_de_empresa                 object 
 8   Tipo_de_crédito                   object 
 9   Tipo_de_garantía                  object 
 10  Producto de crédito               object 
 11  Plazo de crédito                  object 
 12  Tasa_efectiva_promedio_ponderada  float64
 13  margen_adicional                  float64
 14  Montos_desembolsados              float64
 15  Numero_de_creditos_desembolsados  int64  
 16  Grupo_Etnico                      object 

In [ ]:
df.describe()

Tipo_Entidad  Codigo_Entidad  Tasa_efectiva_promedio_ponderada  \
count  8.192306e+06    8.192306e+06                      8.192306e+06   
mean   1.595123e+00    2.472462e+01                      2.596350e+01   
std    3.065859e+00    2.023323e+01                      1.704147e+01   
min    1.000000e+00    1.000000e+00                      0.000000e+00   
25%    1.000000e+00    7.000000e+00                      0.000000e+00   
50%    1.000000e+00    2.600000e+01                      3.703000e+01   
75%    1.000000e+00    3.900000e+01                      3.825000e+01   
max    3.200000e+01    1.250000e+02                      7.860000e+01   

       margen_adicional  Montos_desembolsados  \
count      8.192306e+06          8.192306e+06   
mean       1.963606e-01          2.152101e+07   
std        1.611435e+00          7.402187e+08   
min       -1.200000e+01          1.000000e-02   
25%        0.000000e+00          2.131000e+05   
50%        0.000000e+00          1.000000e+06   
75%        0.000000e+00          4.075066e+06   
max        3.398000e+01          6.949955e+11   

       Numero_de_creditos_desembolsados   Codigo_CIIU  Codigo_Municipio  
count                      8.192306e+06  8.192306e+06      8.192306e+06  
mean                       1.850998e+01  3.893032e+03      3.402746e+04  
std                        8.207006e+02  3.288060e+03      2.742812e+04  
min                        1.000000e+00  1.000000e+01      5.001000e+03  
25%                        1.000000e+00  9.000000e+01      1.100100e+04  
50%                        1.000000e+00  4.651000e+03      2.300100e+04  
75%                        4.000000e+00  6.630000e+03      6.600100e+04  
max                        3.613090e+05  9.900000e+03      9.977300e+04

vemos la frecuencia de las variables, para ver que no hayan datos fuera de lo real

In [ ]:
frecuencia_sexo = df['Sexo'].value_counts()
frecuencia_sexo

Sexo
Masculino     3883405
Femenino      3315150
No aplica      993750
No binario          1
Name: count, dtype: int64

In [ ]:
frecuencia_Nombre_Entidad=df['Nombre_Entidad'].value_counts()
frecuencia_Nombre_Entidad

Nombre_Entidad
Bancolombia                                              2089653
Banco Davivienda                                         1430166
Banco de Bogotá                                           983172
Tuya                                                      780641
BBVA Colombia                                             429598
Scotiabank Colpatria S.A.                                 340233
Banco Caja Social S.A.                                    320026
Banco Falabella S.A.                                      285693
Banagrario                                                249986
AV Villas                                                 184623
Banco Serfinanza S.A.                                     132460
Itaú                                                      123673
Banco Mundo Mujer S.A.                                    111845
Banco de Occidente                                         89696
Banco Popular                                              81983
Bancoomeva

In [ ]:
frecuencia_tipo_garantia=df['Tipo_de_garantía'].value_counts()
frecuencia_tipo_garantia

Tipo_de_garantía
Sin garantia                                                                           6037573
Garantia idónea o no idónea                                                            1996422
Garantía del fondo agropecuario de garantías (FAG)                                       84507
Garantía  fondo nacional de garantías (FNG) o Fondo de Garantías de Antioquia (FGA)      73804
Name: count, dtype: int64

In [ ]:
frecuencia_tipo_de_tasa=df['Tipo_de_Tasa'].value_counts()
frecuencia_tipo_de_tasa

Tipo_de_Tasa
FS        7916243
DTF        115132
IB6         74673
IB1         32588
OTR         29739
IBR         11599
IB3          8147
DTFE         2708
VUVR         1122
LIB3          121
CEC$           76
CECUVR         58
IBE            48
IPC            29
-DTF           23
Name: count, dtype: int64

In [ ]:
df.head()

Tipo_Entidad          Nombre_Tipo_Entidad  Codigo_Entidad   Nombre_Entidad  \
0             1  BC-ESTABLECIMIENTO BANCARIO               1  Banco de Bogotá   
1             1  BC-ESTABLECIMIENTO BANCARIO              52    Bancamía S.A.   
2             1  BC-ESTABLECIMIENTO BANCARIO              43       Banagrario   
3             1  BC-ESTABLECIMIENTO BANCARIO               7      Bancolombia   
4             1  BC-ESTABLECIMIENTO BANCARIO               1  Banco de Bogotá   

  Fecha_Corte Tipo_de_persona       Sexo Tamaño_de_empresa  \
0  20/10/2023         Natural  Masculino         No aplica   
1  20/10/2023         Natural  Masculino      Microempresa   
2  10/11/2023         Natural  Masculino         No aplica   
3  01/12/2023        Jurídica  No aplica   Mediana empresa   
4  24/11/2023         Natural   Femenino         No aplica   

       Tipo_de_crédito                                   Tipo_de_garantía  \
0              Consumo                                       Sin garantia   
1   Crédito productivo  Garantía  fondo nacional de garantías (FNG) o ...   
2              Consumo                                       Sin garantia   
3  Comercial ordinario                        Garantia idónea o no idónea   
4              Consumo                                       Sin garantia   

   ... margen_adicional Montos_desembolsados  \
0  ...              0.0           2197830.00   
1  ...              0.0           2800000.00   
2  ...              0.0           7969200.00   
3  ...              0.0           3538866.93   
4  ...              0.0            585000.00   

   Numero_de_creditos_desembolsados         Grupo_Etnico  \
0                                11  Sin información (1)   
1                                 1  Sin información (1)   
2                                 4  Sin información (1)   
3                                 5            No aplica   
4                                 1  Sin información (1)   

   Antiguedad_de_la_empresa  Tipo_de_Tasa  \
0              No aplica(1)            FS   
1            más de 10 años            FS   
2              No aplica(1)            FS   
3                0 a 5 años            FS   
4              No aplica(1)            FS   

                    Rango_monto_desembolsado                Clase_deudor  \
0                              Hasta 1 SMLMV        Deudor de la entidad   
1  Mayor a 2 SMLMV y menor o igual a 3 SMLMV        Deudor de la entidad   
2    Mayor a 1 SMLMV menor o igual a 3 SMLMV        Deudor de la entidad   
3    Mayor a 1 SMLMV menor o igual a 3 SMLMV        Deudor de la entidad   
4                              Hasta 1 SMLMV  Deudor nuevo en la entidad   

  Codigo_CIIU Codigo_Municipio  
0        10.0           5172.0  
1       143.0          47268.0  
2       150.0          41676.0  
3      8544.0          11001.0  
4      4711.0          70823.0  

[5 rows x 23 columns]

observamos que Grupo_Etnico hay varias filas sin informacion, borramos la colunma

In [ ]:
df = df.drop('Grupo_Etnico', axis=1)

In [ ]:
frecuencia_antiguedad_en_la_empresa=df.Antiguedad_de_la_empresa.value_counts()
frecuencia_antiguedad_en_la_empresa

Antiguedad_de_la_empresa
No aplica(1)                7151318
más de 10 años               641235
más de 5 y hasta 10 años     265271
0 a 5 años                   134482
Name: count, dtype: int64

In [ ]:
frecuencia_tipo_de_persona=df.Tipo_de_persona.value_counts()
frecuencia_tipo_de_persona

Tipo_de_persona
Natural     7198556
Jurídica     993750
Name: count, dtype: int64

In [ ]:
df.head()

Tipo_Entidad          Nombre_Tipo_Entidad  Codigo_Entidad   Nombre_Entidad  \
0             1  BC-ESTABLECIMIENTO BANCARIO               1  Banco de Bogotá   
1             1  BC-ESTABLECIMIENTO BANCARIO              52    Bancamía S.A.   
2             1  BC-ESTABLECIMIENTO BANCARIO              43       Banagrario   
3             1  BC-ESTABLECIMIENTO BANCARIO               7      Bancolombia   
4             1  BC-ESTABLECIMIENTO BANCARIO               1  Banco de Bogotá   

  Fecha_Corte Tipo_de_persona       Sexo Tamaño_de_empresa  \
0  20/10/2023         Natural  Masculino         No aplica   
1  20/10/2023         Natural  Masculino      Microempresa   
2  10/11/2023         Natural  Masculino         No aplica   
3  01/12/2023        Jurídica  No aplica   Mediana empresa   
4  24/11/2023         Natural   Femenino         No aplica   

       Tipo_de_crédito                                   Tipo_de_garantía  \
0              Consumo                                       Sin garantia   
1   Crédito productivo  Garantía  fondo nacional de garantías (FNG) o ...   
2              Consumo                                       Sin garantia   
3  Comercial ordinario                        Garantia idónea o no idónea   
4              Consumo                                       Sin garantia   

   ... Tasa_efectiva_promedio_ponderada margen_adicional  \
0  ...                            39.77              0.0   
1  ...                            52.39              0.0   
2  ...                             0.00              0.0   
3  ...                            38.21              0.0   
4  ...                            38.25              0.0   

   Montos_desembolsados  Numero_de_creditos_desembolsados  \
0            2197830.00                                11   
1            2800000.00                                 1   
2            7969200.00                                 4   
3            3538866.93                                 5   
4             585000.00                                 1   

   Antiguedad_de_la_empresa  Tipo_de_Tasa  \
0              No aplica(1)            FS   
1            más de 10 años            FS   
2              No aplica(1)            FS   
3                0 a 5 años            FS   
4              No aplica(1)            FS   

                    Rango_monto_desembolsado                Clase_deudor  \
0                              Hasta 1 SMLMV        Deudor de la entidad   
1  Mayor a 2 SMLMV y menor o igual a 3 SMLMV        Deudor de la entidad   
2    Mayor a 1 SMLMV menor o igual a 3 SMLMV        Deudor de la entidad   
3    Mayor a 1 SMLMV menor o igual a 3 SMLMV        Deudor de la entidad   
4                              Hasta 1 SMLMV  Deudor nuevo en la entidad   

  Codigo_CIIU Codigo_Municipio  
0        10.0           5172.0  
1       143.0          47268.0  
2       150.0          41676.0  
3      8544.0          11001.0  
4      4711.0          70823.0  

[5 rows x 22 columns]

In [ ]:
# filtrar el banco bancolombia
df_Bancolombia = df[df['Nombre_Entidad'] == 'Bancolombia']

In [ ]:
df_Bancolombia.head()

Tipo_Entidad          Nombre_Tipo_Entidad  Codigo_Entidad Nombre_Entidad  \
3              1  BC-ESTABLECIMIENTO BANCARIO               7    Bancolombia   
5              1  BC-ESTABLECIMIENTO BANCARIO               7    Bancolombia   
9              1  BC-ESTABLECIMIENTO BANCARIO               7    Bancolombia   
12             1  BC-ESTABLECIMIENTO BANCARIO               7    Bancolombia   
21             1  BC-ESTABLECIMIENTO BANCARIO               7    Bancolombia   

   Fecha_Corte Tipo_de_persona       Sexo Tamaño_de_empresa  \
3   01/12/2023        Jurídica  No aplica   Mediana empresa   
5   20/10/2023         Natural   Femenino         No aplica   
9   01/12/2023         Natural   Femenino         No aplica   
12  29/09/2023         Natural  Masculino         No aplica   
21  24/11/2023        Jurídica  No aplica   Mediana empresa   

        Tipo_de_crédito             Tipo_de_garantía  ...  \
3   Comercial ordinario  Garantia idónea o no idónea  ...   
5               Consumo  Garantia idónea o no idónea  ...   
9               Consumo                 Sin garantia  ...   
12              Consumo                 Sin garantia  ...   
21  Comercial ordinario  Garantia idónea o no idónea  ...   

   Tasa_efectiva_promedio_ponderada margen_adicional  Montos_desembolsados  \
3                             38.21              0.0            3538866.93   
5                             39.79              0.0             480291.09   
9                             38.21              0.0             353840.00   
12                             0.00              0.0           14806418.00   
21                            38.26              0.0           76230556.86   

    Numero_de_creditos_desembolsados  Antiguedad_de_la_empresa  Tipo_de_Tasa  \
3                                  5                0 a 5 años            FS   
5                                  2              No aplica(1)            FS   
9                                  1              No aplica(1)            FS   
12                                29              No aplica(1)            FS   
21                                10            más de 10 años            FS   

                       Rango_monto_desembolsado          Clase_deudor  \
3       Mayor a 1 SMLMV menor o igual a 3 SMLMV  Deudor de la entidad   
5                                 Hasta 1 SMLMV  Deudor de la entidad   
9                                 Hasta 1 SMLMV  Deudor de la entidad   
12      Mayor a 1 SMLMV menor o igual a 3 SMLMV  Deudor de la entidad   
21  Mayor a 12 SMLMV y menor o igual a 25 SMLMV  Deudor de la entidad   

   Codigo_CIIU Codigo_Municipio  
3       8544.0          11001.0  
5       8622.0           5154.0  
9       5613.0          17001.0  
12        90.0           5266.0  
21      8010.0          11001.0  

[5 rows x 22 columns]

In [ ]:
#vemos la dimension
df_Bancolombia.shape

(2089653, 22)

In [ ]:
# tamaño de df_Bancolombia
tamano_en_memoria = df_Bancolombia.memory_usage(deep=True).sum()  # Convertir a kilobytes (KB)

# Imprimir el tamaño en memoria
print(f"Tamaño del DataFrame en memoria: {tamano_en_memoria:.2f} MB")

Tamaño del DataFrame en memoria: 2574512913.00 MB


In [ ]:
# Exporta el DataFrame a un archivo CSV
df_Bancolombia.to_csv('datos_Bancolombia.csv', index=False)